packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -1437,6 +1437,25 @@ function getDependency(node) {
     )
   ) {
     return getDependency(node.parent);
+  } else {
+    return stripTailingPropInAssignment(node);
+  }
+}
+
+/**
+ * Assuming () means the passed/returned node:
+ * (props) => (props)
+ * (props.foo) => (props.foo)
+ * (props.foo.bar) = X => (props.foo).bar = X
+ * (props.foo.bar.baz) = X => (props.foo.bar).baz
+ */
+function stripTailingPropInAssignment(node) {
```

Comment:
Just inline in this function into the parent one.

Reply:
👌